cmd/capibmadm/cmd/vpc/key/delete.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package key
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	logf "sigs.k8s.io/cluster-api/cmd/clusterctl/log"
+
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/vpc"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/options"
+)
+
+type keyDeleteOptions struct {
+	name string
+}
+
+// DeleteCommand vpc key delete command.
+func DeleteCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete",
+		Short: "Delete VPC key",
+		Example: `
+# Create key in VPC
+export IBMCLOUD_API_KEY=<api-key>
+capibmadm vpc key delete key-name --region <region>`,
```

Comment:
```suggestion
capibmadm vpc key delete --name <key-name> --region <region>`,
```
lets take key name as input via flag.

/cc @Karthik-K-N @dharaneeshvrd